### ISSUE TITLE
CMake build failures for examples, tests, and miniapps when CUDA is on

### ISSUE THREAD

**Issue by rcarson3** [mfem/mfem#1316]
I came across this issue while trying to build the examples, tests, and miniapps with a CUDA enabled build in order to test out one of the PRs, and I've reproduced the bug with the master branch. As far as I can tell use ```make VERBOSE=1``` the CUDA compiler is not being called when using the appropriate make tests/miniapps/examples call. However, the main library has no problem being made with the CUDA compiler. 

I get an error such as the following out when trying to make those examples:

```
Building CXX object tests/unit/CMakeFiles/unit_tests.dir/unit_test_main.cpp.o
In file included from /g/g20/carson16/test_dir/mfem/general/device.hpp:15:0,
                 from /g/g20/carson16/test_dir/mfem/mfem.hpp:18,
                 from /g/g20/carson16/test_dir/mfem/build/mfem.hpp:3,
                 from /g/g20/carson16/test_dir/mfem/tests/unit/unit_test_main.cpp:13:
/g/g20/carson16/test_dir/mfem/general/cuda.hpp:19:10: fatal error: cuda_runtime.h: No such file or directory
 #include <cuda_runtime.h>
```

I use the following cmake command to initialize everything, and I've had no problem so far using it with my application. I guess for anymore useful information I'm using gcc/7.3.1, CUDA/10.1.243, RAJA v0.11.0, and CMAKE/3.14.5. 

```
 cmake ../ -DMFEM_USE_MPI=ON \
   -DMETIS_DIR=${METIS_DIR} \
   -DMFEM_USE_CONDUIT=ON -DConduit_REQUIRED_PACKAGES=HDF5 -DCONDUIT_DIR=${CONDUIT_DIR} \
   -DHYPRE_DIR=${BASE_DIR}/hypre/src/hypre/ \
   -DHDF5_ROOT:PATH=${HDF5_DIR} \
   -DCMAKE_INSTALL_PREFIX=../install_dir/ \
   -DMFEM_USE_OPENMP=NO \
   -DMFEM_USE_CUDA=YES \
   -DMFEM_USE_RAJA=ON \
   -DRAJA_DIR:PATH=${BASE_DIR}/raja/install_dir/ \
   -DCUDA_ARCH=sm_70 \
   -DCMAKE_BUILD_TYPE=Release
```

**Comment by camierjs** [comment]
We went through this update of the build system in the PR #1104; this will allow you to do want you need.

**Comment by rcarson3** [comment]
@camierjs good to know. I'll go ahead and close this issue then, since it sounds like the fix will be included in the v4.1 release through that PR.
